### DISCUSSION TITLE
Typing for JSON & JSONB columns?

### DISCUSSION THREAD

**Discussion by jvanasco** [sqlalchemy/sqlalchemy#11318] [Usage Questions] [answered]
What is the correct typing for a mapped JSON/JSONB column?  I could not find it in the docs or an example/tests.

Using `Dict` seems to work.  (The following do not work: sqlalchemy.JSON, sqlalchemy.dialects.postgresql.JSON, sqlalchemy.dialects.postgresql.JSONB)

    fingerprint_json: Mapped[Optional[Dict]] = mapped_column(
        sqlalchemy.dialects.postgresql.JSONB, nullable=True
    )

**Comment by CaselIT** [comment, marked as answer]
Hi,

What you please. Any json column can store most python literal type, so the correct typing depend on the use. The most usual one is probably dict thought, but one could use list or str or float or bool
